src/libstd/backtrace.rs

```diff
@@ -159,6 +159,63 @@ enum BytesOrWide {
     Wide(Vec<u16>),
 }
 
+impl fmt::Debug for Backtrace {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut capture = match &self.inner {
+            Inner::Unsupported => return fmt.write_str("unsupported backtrace"),
+            Inner::Disabled => return fmt.write_str("disabled backtrace"),
+            Inner::Captured(c) => c.lock().unwrap(),
+        };
+        capture.resolve();
+
+        write!(fmt, "Backtrace ")?;
+
+        let mut dbg = fmt.debug_list();
+
+        for frame in &capture.frames {
+            dbg.entries(&frame.symbols);
+        }
+
+        dbg.finish()
+    }
+}
+
+impl fmt::Debug for BacktraceSymbol {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut dbg = fmt.debug_struct("");
```

Comment:
I think this trick is resulting in extra spaces where there shouldn't be. Your most recent comment shows:

```console
Backtrace [ {...},  {...},  {...},  {...},  ...,  {...}]
           ^       ^       ^       ^             ^
```

Is there a way to use https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_map instead? Alternatively it would also be fine to emit this without using any of the helpers.

```rust
write!(f, "{ fn: \"{}\"", fn_name)?;
if ... {
    write!(f, ", file: {:?}, line: {}", fname, line)?;
}
write!(f, " }")?;
```